myfirstcatapi/models/cat_model.py

```diff
@@ -171,6 +171,20 @@ def cat_summary_from_bson(cat: BSONDocument) -> dto.CatSummary:
     )
 
 
+async def get_total_cats_count() -> int:
+    cat_list = []
+
+    collection = await get_collection(_COLLECTION_NAME)
+    result = collection.find()
+
+    async for document in result:
+        cat_list.append(document)
+
+    count = len(cat_list)
```

Comment:
I highly advice against doing this to get a count. Imagine we would have 1m+ documents which are very big, then this will definitely introduce memory issues, cause all kind of alerts and in worse case even bring down the entire API service.
If we would use SQL, we also do retrieve all rows just to get a count, right?

Similarly, in mongoDB you can get a count w/o looping through all documents:
```suggestion
    collection = await get_collection(_COLLECTION_NAME)
    count = collection.count_documents()
```
When in doubt, please read the docs:
https://pymongo.readthedocs.io/en/stable/api/pymongo/collection.html#pymongo.collection.Collection.count
^ 
> The count() method is deprecated and not supported in a transaction. Please use count_documents() or estimated_document_count() instead.

So instead:
https://pymongo.readthedocs.io/en/stable/api/pymongo/collection.html#pymongo.collection.Collection.count_documents

Reply:
I have tried this. Even I am using collection.count_documents(), it was giving me error. But now I have removed get_total_count code.